pkg/healthcheck/healthcheck.go

```diff
@@ -608,6 +609,9 @@ func (hc *HealthChecker) runCheck(categoryID CategoryID, c *checker, observer ch
 
 		if err != nil && time.Now().Before(c.retryDeadline) {
 			checkResult.Retry = true
+			checkResult.Err = errors.New("waiting for check to complete")
+			log.Debugf("Retrying on error: %s", err.Error())
```

Comment:
you can do: `log.Debugf("Retrying on error: %s", err)` (we're probably not consistent around the codebase with this)